test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js

```diff
@@ -737,6 +737,132 @@ describe( `[${ host }] Calypso Gutenberg Site Editor Tracking: (${ screenSize })
+						'New header'
+					);
+					await driverHelper.clickWhenClickable( this.driver, createNewHeaderLocator );
+
+					const choosePatternLocator = driverHelper.createTextLocator(
+						By.css( '.wp-block-template-part .block-editor-block-pattern-setup button' ),
+						'Choose'
+					);
+					await driverHelper.clickWhenClickable( this.driver, choosePatternLocator );
+
+					// Wait for this template part to load its new content.
+					await driverHelper.waitUntilElementLocated(
+						this.driver,
+						By.css( `#${ blockId }.block-editor-block-list__layout` )
+					);
+				} );
+
+				const eventsStack = await getEventsStack( this.driver );
+				const createdEvents = eventsStack.filter(
+					( event ) => event[ 0 ] === 'wpcom_block_editor_create_template_part'
+				);
+
+				assert.strictEqual( createdEvents.length, 1 );
+
+				// Verify this doesn't trigger a convert_to event, as they track the same redux action.
+				const convertedEvents = eventsStack.filter(
+					( event ) => event[ 0 ] === 'wpcom_block_editor_convert_to_template_part'
+				);
+				assert.strictEqual( convertedEvents.length, 0 );
+
+				const { variation_slug, content } = createdEvents[ 0 ][ 1 ];
+				assert( variation_slug === 'header' && typeof content === 'string' && content.length > 0 );
+			} );
+
+			it( 'Tracks "wpcom_block_editor_template_part_choose_existing"', async function () {
+				const editor = await SiteEditorComponent.Expect( this.driver );
+				// Undo the template part creation to go back to the placeholder.
+				await driverHelper.clickWhenClickable( this.driver, By.css( 'button[aria-label="Undo"]' ) );
+				await editor.runInCanvas( async () => {
+					const chooseExistingHeaderLocator = driverHelper.createTextLocator(
+						By.css( '.wp-block-template-part.is-selected .components-placeholder button' ),
+						'Choose existing'
+					);
+					await driverHelper.clickWhenClickable( this.driver, chooseExistingHeaderLocator );
+				} );
+
+				await driverHelper.clickWhenClickable(
+					this.driver,
+					By.css( '.wp-block-template-part__selection-preview-item' )
+				);
+
+				const eventsStack = await getEventsStack( this.driver );
+				const chooseEvents = eventsStack.filter(
+					( event ) => event[ 0 ] === 'wpcom_block_editor_template_part_choose_existing'
+				);
+
+				assert.strictEqual( chooseEvents.length, 1 );
+
+				// Verify there are no replace events since these share the same selection component.
+				const replaceEvents = eventsStack.filter(
+					( event ) => event[ 0 ] === 'wpcom_block_editor_template_part_replace'
+				);
+				assert.strictEqual( replaceEvents.length, 0 );
+
+				const { variation_slug, template_part_id } = chooseEvents[ 0 ][ 1 ];
+				// Check the event props, assert id.length > 2 since the format is `{theme}//{slug}`.
+				assert(
+					variation_slug === 'header' &&
+						typeof template_part_id === 'string' &&
+						template_part_id.length > 2
+				);
```

Comment:
nit: Might be easier to follow debugging-wise when checked via separate assertions:
```suggestion
				assert( variation_slug === 'header' );
				assert( typeof template_part_id === 'string' );
				assett( template_part_id.length > 2 );
```